jdaviz/configs/default/plugins/export_plot/export_plot.py

```diff
@@ -20,16 +33,41 @@ class ExportViewer(PluginTemplateMixin, ViewerSelectMixin):
     * :meth:`~jdaviz.core.template_mixin.PluginTemplateMixin.show`
     * :meth:`~jdaviz.core.template_mixin.PluginTemplateMixin.open_in_tray`
     * :meth:`save_figure`
+    * :meth:`save_movie` (Cubeviz only)
     """
     template_file = __file__, "export_plot.vue"
 
+    # For Cubeviz movie.
+    i_start = Any(0).tag(sync=True)
+    i_end = Any(0).tag(sync=True)
+    movie_fps = Any(5.0).tag(sync=True)
+    movie_filename = Any("mymovie.mp4").tag(sync=True)
+    movie_msg = Unicode("").tag(sync=True)
+    movie_recording = Bool(False).tag(sync=True)
+    movie_kill = Bool(False).tag(sync=True)
+
     @property
     def user_api(self):
+        if self.config == "cubeviz":
+            return PluginUserApi(self, expose=('viewer', 'save_figure', 'save_movie'))
         return PluginUserApi(self, expose=('viewer', 'save_figure'))
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
+        if self.config == "cubeviz":
+            if HAS_OPENCV:
+                self.session.hub.subscribe(self, AddDataMessage, handler=self._on_cubeviz_data_added)  # noqa: E501
+            else:
+                # NOTE: HTML tags do not work here.
+                self.movie_msg = 'Please install opencv-python to use this feature.'
+
+    def _on_cubeviz_data_added(self, msg):
+        # NOTE: This needs revising if we allow loading more than one cube.
+        if isinstance(msg.viewer, BqplotImageView):
+            if len(msg.data.shape) == 3:
+                self.i_end = msg.data.shape[-1] - 1  # Same as max_value in Slice plugin

```

Comment:
I wonder if we could have the slice plugin emit a message or have this watch for `slice_plugin.max_value` instead of having to reproduce the logic?

Reply:
This is a one-time set up when data is loaded. To listen to a message the entire time app is running for this one call seems excessive, especially since we only allow one cube to be loaded anyway.

Reply:
This currently listens to any `AddDataMessage` and then just filters to `BqplotImageView` (which will then also be triggered any time that the user loads a moment map or other output into either the flux or uncertainty viewer, but might just then be ignored by the shape check).  My concern is just that this logic is duplicated and we could easily introduce a bug by updating it in one place but not the other.  If you don't think listening to the traitlet in the other plugin makes sense, then I guess at least a comment in both places mentioning the logic-duplication would help prevent mistakes in the future.

Reply:
Do you want me to set this traitlet from the Slice plugin instead? I think I can do that.

Reply:
No, then that would require the slice plugin checking to see if this plugin exists and pushing the value, which for some reason feels more against the whole idea of modular plugins than subscribing to changes 🤷‍♂️.  Whatever you think is best is fine, I won't insist.

Reply:
Let's leave it as-is for now then, since it works. We can always have a follow-up PR to make it cleaner if desired. Thanks. I appreciate your feedback nonetheless.

Reply:
p.s. I will add comment to Slice as you requested.